src/uwtools/drivers/forecast.py

```diff
@@ -72,12 +74,23 @@ def create_model_config(self, config_obj, outconfig_file, base_file=None):
         msg = f"Config file {outconfig_file} created"
         logging.info(msg)
 
-    def stage_fix_files(self):
+    def stage_fix_files(self, run_directory, config_obj):
```

Comment:
Instead of passing in the whole `config_obj`, then pulling the `static` `dict` that we need out of it, could we require the caller to pass in the `dict` that we need? If the caller _has_ the big config object, **they** could pass in `config_obj.get("static", {})`. Or, if they have a mapping of source and destination files they want copied -- wherever they got it -- the could pass that in.

Why should this function know that some dictionary has a `static` key? I think we should start moving away from operating on these large, complex `dict` objects and use specific bits of them as needed.

This should make testing easier, and means that we could write a type-annotated function signature for this function like
```
def stage_files(self, run_directory: str, file_map: Dict[str, str]) -> None:
```
That is, a function that takes a `str` run directory, and a `dict` mapping source to destination (or is it the other way around?) files, where both the keys and values are `str`s.

I left `fix` out of the function name in my example due to my `fix` vs `static` question, below, but also because: Does what this function is doing have anything to do with the files being static? That strikes me as an irrelevant detail here; it might be useful to have in a comment at the call site, like
```
# Stage static files into the run directory:
fv3obj.stage_files(run_directory=rundir, file_map=config.obj.get("static", {}))
```
Could we potentially also use this function to also stage cycle-specific files in the run directory?

Reply:
In conversations with @christinaholtNOAA yesterday, we decided not to include cycle-specific files for now. I do agree, though, that this could be done within the same function.

Reply:
I attempted to add the type-annotated function signature, but it kept failing the linter, so I reverted the changes.

Reply:
Thanks for trying the type signature. I'm planning to give a short tutorial on type annotation sometime in the near future, which I hope will help going forward, but here, specifically, we'd need to `from typing import Dict`, which is what I guess `pylint` complained about. (Quick type geekery: We'd have to annotate the current big config object as `Dict[Any, Any]` (also importing `Any` from `typing`), i.e. `config_obj: Dict[Any, Any]`, which is the same as just `config_obj: dict`, which is saying close to, though not quite, nothing. At least we'd know it's a dictionary, but we'd know nothing about what to expect as content.)

But we'd also need to change type of the argument being passed in, extracting the file mapping from the larger `dict` at the call site. Though I think that's worth doing now, to set a precedent that we can build on, I won't insist. But I think we should talk about whether that's the direction we should go. As a general design principle, I'll assert that functions should not be given more to work with than what they need. And they shouldn't need to have knowledge of things they're not responsible for. The first principle means limits the harm they can do: There's no reason this function should have access to other parts of the config `dict`, but now it could access, and even change, _everything_. The second principle insulates the code from changes it shouldn't have to care about. If we decided to change the key `static` to something else, this code would have to be updated to track that change.

Do we disagree about those ideas? If not, then the question isn't if we should head in that direction, but when. I'm fine with that being "later", but the longer we wait the more debt we accrue.

Reply:
Roger on not dealing with cycle files for now. We could generalize this function's name later if we found it useful for other purposes. As long as we address the `fix` vs `static` question, I'm happy with a specific (and accurate) function name.